pages/pipelines/glossary.md

```diff
@@ -0,0 +1,72 @@
+
+## Artifact
+
+An artifact is a file or metadata generated during a build. You can keep artifacts in a Buildkite-managed storage service or a third-party cloud storage service like Amazon S3, Google Cloud Storage, or Artifactory. Common uses include storing assets like logs and reports, or passing files between steps.
+
+To learn more, see [Build artifacts](/docs/pipelines/artifacts).
+
+## Build
+
+A build is a single run of a pipeline. You can trigger a build in various ways, including through the UI, API, as the result of a webhook, or on a schedule.
+
+## Dynamic pipeline
+
+Dynamic pipelines define their steps at runtime using scripts, giving you the flexibility to only run the steps relevant to particular code changes and workflows.
+
+Dynamic pipelines are helpful when you have a complex build process that requires different steps to execute based on runtime conditions, such as the branch, the environment, or the results of previous steps.
+
+To learn more, see [Dynamic pipelines](/docs/pipelines/defining-steps#dynamic-pipelines).
+
+## Hook
+
+A hook is a method of customizing the behavior of Buildkite through lifecycle events. They let you run scripts at different points of the agent or job lifecycle. Using hooks, you can extend the functionality of Buildkite and automate tasks specific to your workflow and requirements.
+
+To learn more, see [Hooks](/docs/agent/v3/hooks).
+
+## Job
+
+A job is the execution of a command step during a build. Jobs run the commands, scripts, or plugins defined in the step.
+
+## Job queue
+
+The job queue is a queue of jobs waiting to run on an agent. You can configure queues for particular groups of agents, isolating a set of jobs and agents. Typical uses for queues include separating deployment agents and pools of agents for specific pipelines or teams.
+
+To learn more, see [Queues](/docs/agent/v3/queues).
+
+## Pipeline
+
+A pipeline is a container for modeling and defining workflows. They contain a series of steps to achieve goals like building, testing, and deploying software.
+
+To learn more, see the [Pipeline overview](/docs/pipelines).
+
+## Plugin
+
+Plugins are small, self-contained pieces of extra functionality that help you customize Buildkite to your specific workflow. They modify command steps using hooks to perform actions like checking code quality, deploying to cloud services, or sending notifications.
+
+Plugins can be open source and available for anyone to use or private for just your organization.
+
+To learn more, see [Plugins](/docs/plugins).
+
+## Step
+
+A step describes a single, self-contained task as part of a pipeline. You define a step in the pipeline configuration using one of the following types:
+
+- **Command step:** Runs one or more shell commands on one or more agents.
+- **Block step:** Pauses a build until unblocked.
+- **Wait step:** Pauses a build until all previous jobs have completed.
+- **Input step:** Pauses a build while collecting information from a user.
```

Comment:
Close, but I think "pauses" might not be true here. The difference between "block" and "input" steps is that an input step _doesn't_ pause the pipeline by default, only steps which directly depend on it. Both can have fields which collect information.

Reply:
Good catch! Thanks. I think I'm focussing on the wrong bit here, which causes the issue. The most important part of an input step is that it collects information from a user. I'll change it to that and skip the details about what it pauses on this page. Let me know what you think.